packages/api/src/external/ehr/athenahealth/command/get-patient.ts

```diff
@@ -8,104 +8,129 @@ import { Patient, PatientDemoData } from "@metriport/core/domain/patient";
   });
   if (existingPatient) {
-    return await getMetriportPatient({ cxId, id: existingPatient.patientId });
+    const metriportPatient = await getMetriportPatientOrFail({
+      cxId,
+      id: existingPatient.patientId,
+    });
+    return metriportPatient.id;
   }
-  if (!athenaUrl) throw new Error("Athenahealth url not defined");
+  if (!athenaUrl) throw new Error("AthenaHealth url not defined");
   const athenaPatient = await getAthenaPatient({
     cxId,
     accessToken,
     baseUrl: athenaUrl,
     patientId: athenaPatientId,
   });
-  if (!athenaPatient) return undefined;
+  if (!athenaPatient) throw new NotFoundError("AthenaHealth patient not found");
   if (athenaPatient.name.length === 0) {
     throw new Error("AthenaHealth patient missing at least one name");
   }
   if (athenaPatient.address.length === 0) {
     throw new Error("AthenaHealth patient missing at least one address");
   }
 
-  const patientDemos = createMetriportPatientDemos(athenaPatient);
+  const patientDemoFilters = createMetriportPatientDemoFilters(athenaPatient);
   const patients: Patient[] = [];
-  const getPatientByDemoWrapperErrors: string[] = [];
+  const getPatientByDemoErrors: string[] = [];
 
   await executeAsynchronously(
-    patientDemos.map(demo => {
-      return { cxId, demo, patients, errors: getPatientByDemoWrapperErrors, log };
+    patientDemoFilters.map(demo => {
+      return { cxId, demo, patients, errors: getPatientByDemoErrors, log };
     }),
     getPatientByDemo,
     { numberOfParallelExecutions: 5 }
   );
 
-  if (getPatientByDemoWrapperErrors.length > 0) {
+  if (getPatientByDemoErrors.length > 0) {
     capture.error("Failed to get patient by demo", {
       extra: {
         cxId,
-        patientCreateCount: patientDemos.length,
-        errorCount: getPatientByDemoWrapperErrors.length,
-        errors: getPatientByDemoWrapperErrors.join(","),
+        patientDemoFiltersCount: patientDemoFilters.length,
+        errorCount: getPatientByDemoErrors.length,
+        errors: getPatientByDemoErrors.join(","),
         context: "athenahealth.get-patient",
       },
     });
   }
 
-  const metriportPatient = patients[0];
+  let metriportPatient = patients[0];
   if (metriportPatient) {
     const uniquePatientIds = new Set(patients.map(patient => patient.id));
     if (uniquePatientIds.size > 1) {
       capture.message("AthenaHealth patient mapping to more than one Metriport patient", {
         extra: {
           cxId,
-          patientCreateCount: patientDemos.length,
+          patientDemoFiltersCount: patientDemoFilters.length,
           patientIds: uniquePatientIds,
           context: "athenahealth.get-patient",
         },
         level: "warning",
       });
     }
-    await findOrCreatePatientMapping({
+  } else {
+    const defaultFacility = await getFacilityMappingOrFail({
       cxId,
-      patientId: metriportPatient.id,
-      externalId: athenaPatientId,
+      externalId: "default",
```

Comment:
Is this a special value? Expected somewhere else? Should we have a shared constant for it?

Reply:
Moved to const. Is *is* a special value essentially.